### DISCUSSION TITLE
patcher fails with reboot unit

### DISCUSSION THREAD

**Discussion by tl-tasmota** [Mr-MIBonk/M.I.B._More-Incredible-Bash#469] [General]
Using the aio patcher V7.3.1 fails within the apps/flash script at line 243 $PATCHER -m ifs -v patch --pattern $VOLUME/mod/patch_pattern/patch_2_3_FEC_CP.pattern $STAGE2 | $TEE -i -a $LOG.

The log shows that the patch was made but the LZO compress part or later fails. Fails does mean -> the unit made a reboot. Unit is MHIG (Audi).
 
Any idea what to do ?

###  LogFile
--- Free RAM on unit: 245MB
    It will take about 2 minutes. Please wait...
Patch: from: /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check.ifs
Patch: to:   /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check_patched.ifs
header: compressed_size 0xcc5d84
header: decompressed_size 0x22d7488
checksum: 1.39s
header: compressed checksum b'09d93da8'
header: compressed calc_checksum b'09d93da8'
Patch: Image is compressed with valid checksum: 09D93DA8
valid: 1.396s
Decompress: from: /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check.ifs
Decompress: to:   <buffer>
Decompress: length 36533124 bytes
Decompress: LZO version: 2.10
overall: 2.147s
Decompress: decompress time 0.413
Decompress: write time 0.0
Decompress: Finished
Patching :/usr/apps/MIBRoot in decompressed ifs at offset 0x1698000
Patch: did not find /usr/apps/MIBRoot:FEC1
Patch: found /usr/apps/MIBRoot:FEC2 at 0x5a4aac
Patch: found /usr/apps/MIBRoot:FEC3 at 0x5a4b18
Patch: found /usr/apps/MIBRoot:FEC4 at 0x5a9f5f
Patch: found /usr/apps/MIBRoot:FEC5 at 0x5abd13
Patch: found /usr/apps/MIBRoot:Componentent Protection1 at 0xc2e44
Patch: found /usr/apps/MIBRoot:Componentent Protection2 at 0xc2e44
Patch: found /usr/apps/MIBRoot:Componentent Protection3 at 0xc2f00
Patch: found /usr/apps/MIBRoot:Componentent Protection4 at 0xc2f14
Compress: from: <buffer>
Compress: to:   /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check_patched.ifs
Compress: current size: 36533384
checksum: 0.161s
Compress: Using LZO library version: 2.10

**Comment by Vengo222** [comment]
Hi, i have the same problem like you

**Comment by tl-tasmota** [comment]
Hi - found a solution for me. Found out the problem is the line 243 of apps/flash.
     $PATCHER -m ifs -v patch --pattern $VOLUME/mod/patch_pattern/patch_2_3_FEC_CP.pattern $STAGE2 | $TEE -i -a $LOG
which creates a memory fault while having low memory/heapsize within your unit.   Line 233 makes a memory check which shows in my environment 244MB free .  >240MB seems not to be the right minimum for the next steps - specially the patcher line.

What to do ? The patcher command has a command line parameter "heapsize", which limits the used memory.  I added the commandline parameter  in line 243 :
   
				$PATCHER -X heapsize=200M -m ifs -v patch --pattern  .... 

Also change line 233 to 

                                  if [ $FreeRAM -lt 230 ]; then
                                  

After changing both lines the script runs without errors and successfully patching . In the next step I was able to install the newest maps.

**Reply by Vengo222** [reply]
I am not an expert in Lynux programming, I am doing the patch with the M.I.B._More-Incredible-Bash program, so I don't know how to apply the solution you tell me about.

If you tell me the specific steps to follow and the commands that I have to use and where I have to use them, I will surely be able to apply them successfully.

**Comment by Vengo222** [comment]
Now reading your comment better, what you say is to make the changes that you told me about in the flash file in the APPS folder.... could you put the modified flash file to download

**Comment by Vengo222** [comment]
I am doing the patching with the M.I.B program, which also has a flash file in the app folder, it is really the same program that you were using, and the solution will surely be the same, I am attaching a video showing what has happened since I press the patch button until the navi restarts, after the restart it goes to a normal screen without having patched anything... Tell me if it is the same thing that happened to you, the solution is surely the same.  

video: https://drive.google.com/file/d/17Gl_X9xouNuxaZwOqROdC3k2QxfL-wE1/view?usp=sharing

file:  https://drive.google.com/file/d/1NOJ5-AnMkVJzevOhxbFKBuoeCBa2BpL5/view?usp=sharing

**Reply by tl-tasmota** [reply]
Hi Vengo, Had a look to your video/files. Your shared flash file is version **0.3.5** which is not the newest one. Are you shure that you have the newest copy of MIB GitRepo version 3.7.1 ? Look into the /VERSION file of your SD CARD. It should contain : _pre-release V3.7.1 "MHIG Edition"_

**Reply by Vengo222** [reply]
[README.md](https://github.com/user-attachments/files/16134397/README.md)

It is the current "pre-release V3.7.1 "MHIG Edition" and the flash file is indicated as: revision="flash v0.3.7 (2023-07-23 by MIBonk & MIB-Wiki)"

**Comment by Vengo222** [comment]
Do you think your solution can work for my system?

I don't see that it is dangerous, it only modifies the free memory value originally established, if you send me that modified file I can test it on my system.

**Comment by Vengo222** [comment]
In the first of your messages you say that you are using "aio patcher V7.3.1", I don't know that type of patch or program.

I also don't know how to check the system's free memory, and I don't know how you could check it,

Can it be checked with MIB or with Toolbox?

**Comment by Vengo222** [comment]
I just checked the LOG file that is created on the SD card, I attach the files of the last 2 attempts, I think the problem is the same as yours, so the solution must be the same, go to the flash file that you used and I will copy it from the app folder

[MU0415-JTB-07401.04.16777700A7.log](https://github.com/user-attachments/files/16141984/MU0415-JTB-07401.04.16777700A7.log)

[MU0415-JTB-07401.04.16777700A7.log](https://github.com/user-attachments/files/16141987/MU0415-JTB-07401.04.16777700A7.log)

**In both the result is the same and I see that it is the same as the log you posted**
      ---------------------
MMX-flash-done!
--- Reading patch DB...
--- Searching offset and size of root-stage2.ifs...:
[/net/mmx/fs/sda0/apps/sbin/flashlock] Info : Root-Block at Offset (14)!
ifs-root-stage2.ifs offset: 0x00ba0000
ifs-root-stage2.ifs size: 845DCC00
--- OK: OFFSETPART2 in DB and RCC flash match.
--- OK: STAGE2SIZE in DB and RCC flash match.
--- Dumping root-stage2.ifs from RCC flash.
    It will take about 1 minute. Please wait...
--- Patching dumped root-stage2.ifs.
    It will take about 2 minutes. Please wait...
Patch: from: /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check.ifs
Patch: to:   /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check_patched.ifs
header: compressed_size 0xcc5d84
header: decompressed_size 0x22d7488
checksum: 1.231s
header: compressed checksum b'09d93da8'
header: compressed calc_checksum b'09d93da8'
Patch: Image is compressed with valid checksum: 09D93DA8
valid: 1.235s
Decompress: from: /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check.ifs
Decompress: to:   <buffer>
Decompress: length 36533124 bytes
Decompress: LZO version: 2.10
overall: 1.37s
Decompress: decompress time 0.435
Decompress: write time 0.0
Decompress: Finished
Patching :/usr/apps/MIBRoot in decompressed ifs at offset 0x1698000
Patch: did not find /usr/apps/MIBRoot:FEC1
Patch: found /usr/apps/MIBRoot:FEC2 at 0x5a4aac
Patch: found /usr/apps/MIBRoot:FEC3 at 0x5a4b18
Patch: found /usr/apps/MIBRoot:FEC4 at 0x5a9f5f
Patch: found /usr/apps/MIBRoot:FEC5 at 0x5abd13
Patch: found /usr/apps/MIBRoot:Componentent Protection1 at 0xc2e44
Patch: found /usr/apps/MIBRoot:Componentent Protection2 at 0xc2e44
Patch: found /usr/apps/MIBRoot:Componentent Protection3 at 0xc2f00
Patch: found /usr/apps/MIBRoot:Componentent Protection4 at 0xc2f14
Compress: from: <buffer>
Compress: to:   /net/mmx/fs/sda0/patches/9611-ifs-root-stage2-0x00ba0000-845DCC00_check_patched.ifs
Compress: current size: 36533384
checksum: 0.145s
Compress: Using LZO library version: 2.10

**Comment by tl-tasmota** [comment]
Attached the modfied apps/flash file. 

Unzip and copy( = replace)  to SD - Card.
[flash.zip](https://github.com/user-attachments/files/16148159/flash.zip)

**Reply by Vengo222** [reply]
Thanks, I'll try it on Friday.

**Reply by tl-tasmota** [reply]
do you have success ?
